New package: FluxPruning v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: FluxPruning
- Repository: https://github.com/EBothereau/FluxPruning.jl
- Created by: @EBothereau
- Version: v0.1.0
- Commit: aa7d2ecaf5ff64d08b633fb860b4232d6224f74d
- Git reference: HEAD

giordano commented on F/FluxPruning/Deps.toml:
```diff
@@ -0,0 +1,14 @@
+[0]
+CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
+ChainRulesCore = "d360d2e6-b24c-11e9-a2a3-2a2ae2dbcce4"
+Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+DocumenterTools = "35a29f4d-8980-5a13-9543-d66fff28ecb8"
```
Doesn't look Documenter is used anywhere in the code of your package: https://github.com/search?q=repo%3AEBothereau%2FFluxPruning.jl%20documenter&type=code.  That shouldn't be a dependency.

giordano commented on F/FluxPruning/Deps.toml:
```diff
@@ -0,0 +1,14 @@
+Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+RandomNumbers = "e6cf234a-135c-5ec9-84dd-332b85af5143"
+Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
```
And test is a test-only dependency: https://github.com/search?q=repo%3AEBothereau%2FFluxPruning.jl+Test+language%3AJulia&type=code&l=Julia

goerz commented:
I agree, definitely make sure to only keep the dependencies of the code itself in the `Project.toml` file, and manage test or docs dependencies separately.

I presume that is what you were trying to do in https://github.com/JuliaRegistries/General/pull/126418. But you don't have to change the version number for that. If you make changes, and re-register the new commits with the same version number `0.1.0`, it will update this existing PR. That would be the way to go in situation like this.

Give me a quick ping once you have the dependencies updated, so I can unblock the registration.

EBothereau commented:
@goerz , Yes, I'm really sorry about the multiple branches, I learned a lot today. I removed Test and moved it to [extras]. Is that what I was supposed to do (I could not find more detailed explanations)?

 I think this is indeed the last release with the test change.

lgoettgens commented on F/FluxPruning/Compat.toml:
```diff
@@ -0,0 +1,8 @@
+Functors = "0.4"
+RandomNumbers = "1.6.0 - 1"
+Zygote = "0.6"
+julia = "1.6 - 1.11"
```
Please don't upper bound the julia version (unless you have a very serious reason to do so). You package will very likely just work on 1.12 once that gets released.
So what I am suggesting here is to replace `julia = "1.6 - 1.11"` in your Project.toml by `julia = "1.6"` (and re-trigger registration).

(And there is a typo in your mail address in the Project.toml you might wanna fix, but that's not relevant for this registration.)

EBothereau replied:
I've made the necessary changes, thank you ! 

[noblock]

JuliaHubRegistrator committed:
New package: FluxPruning v0.1.0

UUID: aab75b78-9f0d-402c-967e-59140ef06c24
Repo: https://github.com/EBothereau/FluxPruning.jl.git
Tree: 040e1be912ce0f56f874efa68404ba1230eeb7db

Registrator tree SHA: 191228b6dd8b9d0e2965ae3e705fe54c51dcfee8
